ui/apps/ui/src/app/components/download-results/download-results-button.component.ts

```diff
@@ -0,0 +1,151 @@
+    <div
+      class="download-button"
+      (click)="downloadResults()"
+      [ngbTooltip]="'Download first 2000 results as a CSV file'"
+    >
+      <img
+        class="download-icon"
+        src="assets/download-button.svg"
+        alt="download-icon"
+      />
+      Download Results
+    </div>
+
+    <ng-template #content let-modal>
+      <div class="modal-header download-results-title">
+        <h4 class="modal-title">Downloading CSV file</h4>
+      </div>
+      <div class="modal-body">
+        <div class="container mb-4">
+          <div style="text-align: center;">
+            <ng-container *ngIf="modalLoading">
+              <nz-spin nzSimple nzSize="large"></nz-spin>
+            </ng-container>
+            <ng-container *ngIf="modalError && !modalLoading">
+              Failed to fetch the results. Please try again later.
+            </ng-container>
+          </div>
+        </div>
+      </div>
+    </ng-template>
+  `,
+  styles: [
+    `
+      .download-button {
+        font-family: 'Inter', serif;
+        font-style: normal;
+        font-weight: 500;
+        font-size: 16px;
+        line-height: 20px;
+        color: #3d4db6;
+
+        display: inline-block;
+        cursor: pointer;
+      }
+
+      .download-icon {
+        vertical-align: text-bottom;
+        height: 20px;
+      }
+    `,
+  ],
+})
+export class DownloadResultsButtonComponent {
+  @ViewChild('content') content: any;
+  private modalRef?: NgbModalRef;
+
+  modalLoading = true;
+  modalError = false;
+
+  constructor(
+    private _customRoute: CustomRoute,
+    private _router: Router,
+    private _route: ActivatedRoute,
+    private _fetchDataService: FetchDataService,
+    private _searchMetadataRepository: SearchMetadataRepository,
+    private _adaptersRepository: AdaptersRepository,
+    private _modalService: NgbModal
+  ) {}
+
+  downloadResults() {
+    this.modalOpen();
+    this.modalLoading = true;
+
+    this._customRoute.params$
+      .pipe(
+        first(),
+        filter(({ collection }) => !!collection),
+        switchMap((routerParams) => {
+          const { collection } = routerParams;
+          const metadata = this._searchMetadataRepository.get(
+            collection
+          ) as ICollectionSearchMetadata;
+
+          let searchMetadata;
+
+          if (routerParams.standard.toString() === 'true') {
+            searchMetadata = {
+              ...constructRegularSearchMetadata(metadata, routerParams),
+              ...metadata.downloadParams,
+              rows: DEFAULT_NO_OF_RESULTS,
+            };
+          } else {
+            searchMetadata = {
+              ...constructAdvancedSearchMetadata(metadata, routerParams),
+              ...metadata.downloadParams,
+              rows: DEFAULT_NO_OF_RESULTS,
+            };
+          }
+
+          return this._fetchDataService
+            .downloadResults$(searchMetadata, metadata.facets)
+            .pipe(untilDestroyed(this));
```

Comment:
This is only single shot observable (similar to promise) `untilDestroyed` is not necessary here